pkg/cluster/cache_whitebox_test.go

```diff
@@ -392,19 +336,19 @@ func TestRefreshCache(t *testing.T) {
 
 func TestMultipleActionsInParallel(t *testing.T) {
 	// given
+	clusterForTest := newTestCachedToolchainCluster(t, "clusterForTest", ready)
+
 	defer resetClusterCache()
 	var latch sync.WaitGroup
 	latch.Add(1)
 	var waitForFinished sync.WaitGroup
 
-	memberCluster := newTestCachedToolchainCluster(t, "memberCluster", Member, ready)
-	hostCluster := newTestCachedToolchainCluster(t, "hostCluster", Host, ready)
 	clusterCache.refreshCache = func() {
-		clusterCache.addCachedToolchainCluster(memberCluster)
-		clusterCache.addCachedToolchainCluster(hostCluster)
+		clusterCache.addCachedToolchainCluster(clusterForTest)
```

Comment:
we should use more than one toolchaincluster resource I guess, so we can ensure it can work with multiple ones using concurrency.

Reply:
We can keep this for follow up PRs, let's maybe think about this later as I've just discussed with Matous, it may be tricky to handle such a scenario.